ember-file-upload/src/system/upload.ts

```diff
@@ -105,15 +105,22 @@ export function upload(
   file.state = FileState.Uploading;
 
   return waitForPromise(
-    uploadFn(request, options).then(
-      function (response) {
-        file.state = FileState.Uploaded;
+    uploadFn(request, options)
+      .then(function (response) {
+        file.queue?.uploadSucceeded(file, response);
+        // Don't update file.state if it was updated in the callback
+        if (file.state === FileState.Uploading) {
+          file.state = FileState.Uploaded;
+        }
```

Comment:
Regarding (2)

Reply:
Why would a user update the file state?

Reply:
Good point. I suppose I just guessed that they might for some unknown reason. Like how some users update the file name.

I think the `state` setter might be private as only the getter is documented public API.